itests/org.openhab.binding.mielecloud.tests/src/main/java/org/openhab/binding/mielecloud/internal/config/servlet/CreateBridgeServletTest.java

```diff
@@ -74,11 +74,10 @@ public void whenBridgeReconfigurationFailsDueToMissingBridgeThenAWarningIsShownO
         assertNotNull(createBridgeServlet);
 
         Inbox inbox = mock(Inbox.class);
-        when(inbox.add(any())).thenReturn(false);
         setPrivate(Objects.requireNonNull(createBridgeServlet), "inbox", inbox);
 
         ThingRegistry thingRegistry = mock(ThingRegistry.class);
-        when(thingRegistry.get(any())).thenReturn(null);
+        when(thingRegistry.get(any())).thenThrow(new BridgeReconfigurationFailedException(""));
```

Comment:
`ThingRegistry` will never throw `BridgeReconfigurationFailedException`, so this shouldn't be done in tests.

Reply:
It just mocks the behavior before: if inbox.add returned false and thingRegistry.get returned null this exception is thrown.

Reply:
It's not about changing the test behavior. Mocking `ThingRegistry.get` this way breaks the method contract of  `ThingRegistry.get`, i.e. the code doesn't behave the same way when executing tests as when running it productively. In consequence this might lead to subtle bugs when code is changed that calls this method. It might seem fine now but can easily break in the future.